web/src/clipboard_handler.ts

```diff
@@ -0,0 +1,45 @@
+import * as blueslip from "./blueslip";
+import {try_stream_topic_syntax_text} from "./copy_and_paste";
+
+export function clipboard_handler($link_element: JQuery, hide_popover: () => void): void {
+    if (typeof $link_element.data("clipboard-text") === "string") {
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        const stream_topic_link_element_data = $link_element.data("clipboard-text") as string;
+        const stream_topic_syntax_text = try_stream_topic_syntax_text(
+            stream_topic_link_element_data,
+        );
+
+        // invalid links or stream topic names containing markdown cause this variable to be null
+        if (stream_topic_syntax_text === null) {
+            navigator.clipboard
+                .writeText(stream_topic_link_element_data)
+                .then(hide_popover)
+                .catch(() => {
+                    blueslip.error("Clipboard access denied: requires user interaction.");
+                });
```

Comment:
The variable becomes null if the link is invalid. We wouldn't wish to copy an invalid link to clipboard. Let us use assert here too, since we expect it not to be null ever.